include/can_interface.h

```diff
@@ -51,16 +108,17 @@ constexpr typename std::enable_if<std::is_unsigned<T>::value, T>::type bswap(T i
 
 constexpr uint8_t CANSignal_generate_position(uint8_t position, uint8_t length, ICANSignal::ByteOrder byte_order)
 {
-    return (byte_order == ICANSignal::ByteOrder::kLittleEndian
-            || (length - (8 - (position % 8)) /* bits_in_last_byte */ < 0))
-               ? position
-               : position
-                     - ((8
-                         * (((length - (8 - (position % 8)) /* bits_in_last_byte */) % 8) /* remaining_bits */ == 0
-                                ? ((length - (8 - (position % 8)) /* bits_in_last_byte */) / 8) /* full_bytes */
-                                : ((length - (8 - (position % 8)) /* bits_in_last_byte */) / 8) /* full_bytes */ + 1))
-                        + (8 - ((length - (8 - (position % 8)) /* bits_in_last_byte */) % 8) /* remaining_bits */)
-                        - (8 - (position % 8)) /* bits_in_last_byte */);
+    return static_cast<uint8_t>(
+        (byte_order == ICANSignal::ByteOrder::kLittleEndian
+         || (length - (8 - (position % 8)) /* bits_in_last_byte */ < 0))
+            ? position
+            : position
+                  - ((8
+                      * (((length - (8 - (position % 8)) /* bits_in_last_byte */) % 8) /* remaining_bits */ == 0
+                             ? ((length - (8 - (position % 8)) /* bits_in_last_byte */) / 8) /* full_bytes */
+                             : ((length - (8 - (position % 8)) /* bits_in_last_byte */) / 8) /* full_bytes */ + 1))
+                     + (8 - ((length - (8 - (position % 8)) /* bits_in_last_byte */) % 8) /* remaining_bits */)
+                     - (8 - (position % 8)) /* bits_in_last_byte */));
     /*
         if (byte_order == ICANSignal::ByteOrder::kLittleEndian)
```

Comment:
I don't think we should - it's the same logic as the incomprehensible very long return statement, and should be kept for clarity since it's hard enough to understand even with the commented code

Reply:
nit: maybe leave a comment just to say that? not sure if it's confusing for people who are seeing this codebase for the first time